### PR TITLE
Ability to configure the get method

### PR DESCRIPTION
**Pull request by lukemarsh**
I noticed you had a createMethod and an updateMethod so I have added a getMethod. I am using an API which is not particularly RESTful and they only use POST! I have also created a pull request for ng-admin too.

### CODE REVIEW

File: lib/Entity/Entity.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ class Entity {
         this._url = null;
         this._createMethod = null; // manually set the HTTP-method for create operation, defaults to post
         this._updateMethod = null; // manually set the HTTP-method for update operation, defaults to put
+        this._getMethod = null; // manually set the HTTP-method for the get operation, defaults to get
```
Comment (fzaninotto) [inline lib/Entity/Entity.js]: Call it `_retrieveMethod` instead, for better compatibility with the CRUD acronym. By the way, it's probably time to add `_deleteMethod`, too.

**Comment by fzaninotto** [comment]
Also, please add tests to `EntityTest` and `ReadQueriesTest`

**Comment by lukemarsh** [comment]
I have renamed the getMethod and added a deleteMethod. I have added some tests in EntityTest but I couldn't find any tests for passing in custom methods hence why there are no tests there :)

File: tests/lib/Entity/EntityTest.js [hunk 2]
```diff
@@ -87,6 +87,30 @@ describe('Entity', function() {
...
+
+    describe('deleteMethod', function() {
+        it('should return given deleteMethod if already set', function() {
+            var post = new Entity('post').deleteMethod('post');
+            assert.equal('post', post.deleteMethod());
+        });
+
+        it('should return null if no deleteMethod has been set', function() {
+            var post = new Entity('post');
+            assert.equal(null, post.deleteMethod());
```
Comment (jpetitcolas) [inline tests/lib/Entity/EntityTest.js]: Shouldn't we set `DELETE` as a default?

File: tests/lib/Entity/EntityTest.js [hunk 3]
```diff
@@ -87,6 +87,30 @@ describe('Entity', function() {
...
 
+    describe('retrieveMethod', function() {
+        it('should return given retrieveMethod if already set', function() {
+            var post = new Entity('post').retrieveMethod('post');
+            assert.equal('post', post.retrieveMethod());
+        });
+
+        it('should return null if no retrieveMethod has been set', function() {
+            var post = new Entity('post');
+            assert.equal(null, post.retrieveMethod());
```
Comment (jpetitcolas) [inline tests/lib/Entity/EntityTest.js]: Same

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**